clang/lib/AST/ASTContext.cpp

```diff
@@ -3363,6 +3363,7 @@ static void encodeTypeForFunctionPointerAuth(const ASTContext &Ctx,
 #include "clang/Basic/RISCVVTypes.def"
       llvm_unreachable("not yet implemented");
     }
+    break;
```

Comment:
Does inserting an `llvm_unreachable()` here work instead?

Reply:
Would it break if the case is not presented if we change to `llvm_unreachable()`?

Reply:
I mean, it's not supposed to crash if the case is not presented in the original semantic right?

Reply:
It would crash anyway if we fell through because the first thing done in `Type::Record` is a `getAs<RecordType>()` followed by `->` which will dereference a null pointer when handed a `BuiltinType`.

I believe the intent here is that all the cases in the nested `switch` are expected to return and that we handle all of the builtin types.

Reply:
I guess it won't get into `Type::Record` if it fall through from `Type::Builtin`, however I agree with you the intention of the code snippet.
Let me add a default case which is unreachable.

Reply:
won't you get an error for having a `default` case on a fully covered switch? Plus that will remain working if someone adds a new enum value, but doesn't update this code, right?

Reply:
My thinking was: if we add the `llvm_unreachable` as the last statement in the case for `Type::Builtin`, then any case in the nested switch which does not return will eventually hit that unreachable and be caught as a bug rather than falling through. If someone adds a new case to the nested switch, this still happens.

As for the diagnostic, it's off by default (https://godbolt.org/z/PPs993TT9) but is something we explicitly opt into for the project to help us catch bugs: https://github.com/llvm/llvm-project/blob/bee2654300a8f524e05dd4cad41411d597246ac0/llvm/cmake/modules/HandleLLVMOptions.cmake#L824

Reply:
I also agree with you that it should not reach the end of the case for `Type::Builtin`.